### PR TITLE
update to winit 0.29.4

### PR DESCRIPTION
**Pull request by pillowtrucker**
## Checklist

- CI Checked:
  - [ √] `cargo fmt` has been ran
  - [√] `cargo clippy` reports no issues
  - [ √] `cargo test` succeeds
  - [ √] `cargo rend3-doc` has no warnings
  - [ ] `cargo deny check` issues have been fixed or added to deny.toml
 
 ```
  error[duplicate]: found 3 duplicate entries for crate 'windows_x86_64_msvc'
    ┌─ /home/wrath/BRAINWORMS-depends/rend3/Cargo.lock:395:1
    │  
395 │ ╭ windows_x86_64_msvc 0.42.2 registry+https://github.com/rust-lang/crates.io-index
396 │ │ windows_x86_64_msvc 0.48.5 registry+https://github.com/rust-lang/crates.io-index
397 │ │ windows_x86_64_msvc 0.52.0 registry+https://github.com/rust-lang/crates.io-index
    │ ╰────────────────────────────────────────────────────────────────────────────────^ lock entries
```
I don't know how to fix that or understand why that's a problem.
- Manually Checked:
  - [√ ] relevant examples/test cases run
  All the examples other than https://github.com/BVE-Reborn/rend3/issues/543
  - [ √] changes added to changelog
    - [ √] Add credit to yourself for each change: `Added new functionality @githubname`.

## Description
Updated this to winit 0.29.4 since egui ran the update yesterday and dependabot is pestering me

### CODE REVIEW

**Comment by pillowtrucker** [comment]
After fixing up wasm build (noticed on my own branch ci..) Clippy now seems to be demanding the impossible, or at least I'm not sure how to resolve it because the preprocessor blocks are too smart for me and won't let me stick a cheeky return statement or any other hack that I could think of in there based on target arch
```
warning: this let-binding has unit value
   --> rend3-framework/src/lib.rs:306:5
    |
306 | /     let _ = winit_run(event_loop, move |event, event_loop_window_target| {
307 | |         let event = match event {
308 | |             Event::UserEvent(UserResizeEvent::Resize { size, window_id }) => Event::WindowEvent {
309 | |                 window_id,
...   |
363 | |         )
364 | |     });
    | |_______^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_unit_value
    = note: `#[warn(clippy::let_unit_value)]` on by default
help: omit the `let` binding
```
If I omit the let binding it will fail on the non-wasm build demanding that I do something with the Result<>

**Comment by cwfitzgerald** [comment]
Take a look at https://github.com/gfx-rs/wgpu/blob/trunk/examples/src/framework.rs#L378-L390 where I solved that problem in the wgpu exampels

File: examples/cube-no-framework/src/main.rs [hunk 1]
```diff
@@ -176,76 +176,82 @@ fn main() {
...
+                    &renderer.device,
+                    preferred_format,
+                    glam::UVec2::new(resolution.x, resolution.y),
+                    rend3::types::PresentMode::Fifo,
+                );
+                // Tell the renderer about the new aspect ratio.
+                renderer.set_aspect_ratio(resolution.x as f32 / resolution.y as f32);
+            }
+            // Render!
+            winit::event::Event::AboutToWait => {
```
Comment (cwfitzgerald) [inline examples/cube-no-framework/src/main.rs]: We should use redraw requested for this event

File: rend3-framework/src/lib.rs [hunk 2]
```diff
@@ -208,7 +214,9 @@ pub async fn async_start<A: App<T> + 'static, T: 'static>(mut app: A, window_bui
     app.register_panic_hook();
 
     // Create the window invisible until we are rendering
-    let (event_loop, window) = app.create_window(window_builder.with_visible(false));
+    let Ok((event_loop, window)) = app.create_window(window_builder.with_visible(false)) else {
+        exit(1)
```
Comment (cwfitzgerald) [inline rend3-framework/src/lib.rs]: Can't this just be an unwrap?

**Review by cwfitzgerald — CHANGES_REQUESTED** [review]
I have some comments, but I'm just going to address them myself. Looks good in general though, thank you!

**Review by cwfitzgerald — APPROVED** [review]

**Pull request merged**